### DISCUSSION TITLE
How to get Nextjs deployment timestamp on frontend?

### DISCUSSION THREAD

**Discussion by curiosbasant** [vercel/community#574] [Help] [answered]
I want to show a timestamp that tells, when the website _(built in nextjs)_ was last updated.
I hv hosted my nextjs project on vercel.

Can I get the timestamp probably from a env variables or something?

**Comment by swarnava** [comment, marked as answer]
Using [preval.macro](https://github.com/kentcdodds/preval.macro), you can compute the build date and store it as a constant in your app:

```js
import preval from 'preval.macro'

const buildDate = preval`module.exports = new Date().toDateString()`
```

you could even format the date fancier with a module:

```js
import preval from 'preval.macro'

const buildDate = preval`
import format from 'date-fns/format'
module.exports = format(new Date(), 'YYYY-MM-DD')
```
